### PR TITLE
III-6541 bugfix nulls

### PR DESCRIPTION
**Pull request by JonasVHG**
### Changed

- `CachedUserIdentityResolver`: prevent caching `Null` by throwing an catching an `InvalidArgumentException` & refactor to improve readability.
- `CachedUserIdentityResolverTest`: Add unit tests to check `Null` isn't stored in the cache.

### Fixed

- `Null` is no longer cached.

---

Ticket: https://jira.publiq.be/browse/III-6541

### CODE REVIEW

File: src/User/Keycloak/CachedUserIdentityResolver.php [hunk 1]
```diff
@@ -25,38 +26,68 @@ public function __construct(
...
     {
-        return $this->deserializeUserIdentityDetails(
-            $this->cache->get(
-                $this->createCacheKey($userId, 'user_id'),
-                function () use ($userId) {
-                    return $this->getUserIdentityDetailsAsArray($this->userIdentityResolver->getUserById($userId));
-                }
-            )
-        );
+        try {
```
Comment (JonasVHG) [inline src/User/Keycloak/CachedUserIdentityResolver.php:29]: This function is becoming rather complex. This is something that can/should be tackled when we are on PHP 8

Comment (LucWollants) [reply src/User/Keycloak/CachedUserIdentityResolver.php:29]: Try to throw the `CacheException` inside the `getUserIdentityDetailsAsArray` method and use an arrow function.

**Review by LucWollants — APPROVED** [review]
